src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1081,19 +1081,22 @@ function ($a) {
                         ->arrayNode('buses')
                             ->defaultValue(array('messenger.bus.default' => array('default_middleware' => true, 'middleware' => array())))
                             ->useAttributeAsKey('name')
-                            ->prototype('array')
+                            ->arrayPrototype()
                                 ->addDefaultsIfNotSet()
                                 ->children()
-                                    ->booleanNode('default_middleware')->defaultTrue()->end()
+                                    ->enumNode('default_middleware')
```

Comment:
at this point, what about

```yaml
default_middleware:
  logging: true
  route: true
  allow_no_handlers: false
```

that will everybody understand instantly

Reply:
not convinced this should be done (in this PR at least)